File: NeetCode-150/Arrays and Hashing/Valid Anagram/solution.java
// Sid 
class Solution {
    public boolean isAnagram(String s, String t) {
        int[] freqs = new int[26]; 
        int[] freqt = new int[26]; 

        for(int i = 0; i < s.length(); i++){
            char ch = s.charAt(i); 
            freqs[ch-'a']++;
        }

        for(int i = 0; i < t.length(); i++){
            char ch = t.charAt(i);
            freqt[ch-'a']++; 
        }

        for(int i = 0; i < 26; i++){
            if(freqs[i] != freqt[i])
                return false;
        }
        return true;
    }
}

File: NeetCode-150/Arrays and Hashing/Valid Anagram/solution.cpp
// Hari

class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<char, int> umap;
        
        for (int i = 0; i<s.size(); i++) {
            if (umap.find(s[i]) != umap.end()) {
                umap[s[i]] += 1;
            } else {
                umap[s[i]] = 1;
            }
        }
        
        for (int i = 0; i<t.size(); i++) {
            if (umap.find(t[i]) == umap.end()) {
                // not anagram
                return false;
            } else {
                umap[t[i]] -= 1;
            }
        }
        
        for (auto i = umap.begin(); i != umap.end(); i++) {
            if (i->second != 0) return false; 
        }
        
        return true;
    }
};
